Assets/Scripts/Gameplay/Items/Item.cs

```diff
@@ -0,0 +1,42 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+[RequireComponent(typeof(Collider))]
+public class Item : MonoBehaviour, IInteractable
+{
+    [SerializeField]
+    private ItemSkill m_ItemSkill;
+    public void Interact(ICanInteract interactor)
+    {
+        if (interactor != null && interactor is Player player)
+        {
+            PlayPickUpSound();
+            HandleItemSkill(player);
+            Destroy(gameObject);
+        }
+    }
+    
+    private void PlayPickUpSound()
+    {
+        AudioManager.m_Instance.PlaySound("MAGIC_Powerup", 1.0f, 1.2f);
+    }
+
+    public void HandleItemSkill(Player player)
+    {
+        Transform skillObject = player.transform.Find("Skills");
+        
+        if (skillObject == null)
+        {
+            Debug.LogError("Player should have a child GameObject named Skills");
+            return;
+        }
+        
+        Debug.Log("Added");
+        ItemSkill itemSkill = skillObject.gameObject.AddComponent(m_ItemSkill.GetType()) as ItemSkill; //adds child type
```

Comment:
Just being fussy on naming here. It's named skillObject but you're actually grabbing the transform. And then you call gameObject on it.